### PR TITLE
Training pipeline of NeurIPS2019-Learn-to-Move-Challenge

### PR DESCRIPTION
**Pull request by zenghsh3**

### CODE REVIEW

**Comment by zenghsh3** [comment]
Before merging this PR, we need to release version PARL "1.2.1".

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
- 
+# The Winning Solution for the NeurIPS 2019: Learn to Move Challenge
+
 ## Dependencies
 - python3.6
 - [paddlepaddle>=1.5.2](https://github.com/PaddlePaddle/Paddle)
```
Comment (TomorrowIsAnOtherDay) [inline examples/NeurIPS2019-Learn-to-Move-Challenge/README.md]: paddlepaddle==1.5.2, we are not going to maintain this folder in the future, as it takes a lot of time to validate its performance in the new version.

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md [hunk 2]
```diff
@@ -1,8 +1,9 @@
- 
+# The Winning Solution for the NeurIPS 2019: Learn to Move Challenge
```
Comment (TomorrowIsAnOtherDay) [inline examples/NeurIPS2019-Learn-to-Move-Challenge/README.md]: Please introduce the aim of this folder at the beginning:
The **PARL** team gets the first place in [Neurips reinforcement learning competition](https://www.aicrowd.com/challenges/neurips-2019-learn-to-move-walk-around), again! This folder contains our final submitted model and the code relative to the training process.

File: examples/NeurIPS2019-Learn-to-Move-Challenge/opensim_model.py [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        super(CriticModel, self).__init__()
+        hid0_size = 800
+        hid1_size = 400
+        vel_hid0_size = 200
+        vel_hid1_size = 400
+
+        self.obs_dim = obs_dim
+        self.vel_obs_dim = vel_obs_dim
+
+        scope_name = 'critic'
```
Comment (TomorrowIsAnOtherDay) [inline examples/NeurIPS2019-Learn-to-Move-Challenge/opensim_model.py:108]: Why are we using the scope name here? 
Now it takes **5 lines** to create a fc layer.

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md [hunk 4]
```diff
@@ -11,8 +12,80 @@
...
+
+
+## Part2: Curriculum learning
+
+#### 1. Run as fast as possible -> run at 3.0 m/s -> walk at 2.0 m/s -> walk slowly at 1.3 m/s
+The curriculum learning pipeline to get a walking slowly model is the same pipeline in [our winning solution in NeurIPS 2018: AI for Prosthetics Challenge](https://github.com/PaddlePaddle/PARL/tree/develop/examples/NeurIPS2018-AI-for-Prosthetics-Challenge). You can get a walking slowly model by following the [guide](https://github.com/PaddlePaddle/PARL/tree/develop/examples/NeurIPS2018-AI-for-Prosthetics-Challenge#part2-curriculum-learning).
+
+We also provide a pre-trained model that walk naturally at ~1.3m/s. You can download the model file (naming `low_speed_model`) from online stroage service: [Baidu Pan](https://pan.baidu.com/s/1Mi_6bD4QxLWLdyLYe2GRFw) (password: `q9vj`) or [Google Drive](https://drive.google.com/file/d/1_cz6Cg3DAT4u2a5mxk2vP9u8nDWOE7rW/view?usp=sharing).
+
+#### 2. difficulty=1
```
Comment (TomorrowIsAnOtherDay) [inline examples/NeurIPS2019-Learn-to-Move-Challenge/README.md:33]: I understand that it takes a lot of stages to train such as successful agent, but can you write a shell script for these commands?
I'm afraid that there is too much code at the READEME at the moment.

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md [hunk 5]
```diff
@@ -11,8 +12,80 @@
...
+At first, start a local cluster with 300 CPUs:
+
+```bash
+xparl start --port 8010 --cpu_num 300
```
Comment (TomorrowIsAnOtherDay) [inline examples/NeurIPS2019-Learn-to-Move-Challenge/README.md:52]: By the way, we could tell user how to use xparl at the beginning.

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md [hunk 6]
```diff
@@ -11,8 +12,80 @@
...
+```
+
+Note: If there are not enough CPUs in your local machine, you can add CPUs resource of other machines to the cluster by running the command:
+```bash
+xparl connect --address [CLUSTER_IP]:8081
```
Comment (TomorrowIsAnOtherDay) [inline examples/NeurIPS2019-Learn-to-Move-Challenge/README.md]:
```shell
# master
xparl start --port 8010
# other nodes
xparl connect --address [CLUSTER_IP]:8081
```

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md [hunk 7]
```diff
@@ -1,8 +1,9 @@
- 
+# The Winning Solution for the NeurIPS 2019: Learn to Move Challenge
+
```
Comment (TomorrowIsAnOtherDay) [inline examples/NeurIPS2019-Learn-to-Move-Challenge/README.md:6]: Could you add an image demonstrating the performance? I will send you the gif.

**Review by TomorrowIsAnOtherDay — CHANGES_REQUESTED** [review]

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md (hunk 1 above)
Comment (zenghsh3) [reply examples/NeurIPS2019-Learn-to-Move-Challenge/README.md]: done

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md (hunk 2 above)
Comment (zenghsh3) [reply examples/NeurIPS2019-Learn-to-Move-Challenge/README.md]: done

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md (hunk 4 above)
Comment (zenghsh3) [reply examples/NeurIPS2019-Learn-to-Move-Challenge/README.md:33]: done

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md (hunk 5 above)
Comment (zenghsh3) [reply examples/NeurIPS2019-Learn-to-Move-Challenge/README.md:52]: done

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md (hunk 6 above)
Comment (zenghsh3) [reply examples/NeurIPS2019-Learn-to-Move-Challenge/README.md]: done

File: examples/NeurIPS2019-Learn-to-Move-Challenge/README.md (hunk 7 above)
Comment (zenghsh3) [reply examples/NeurIPS2019-Learn-to-Move-Challenge/README.md:6]: done

**Review by TomorrowIsAnOtherDay — APPROVED** [review]

**Pull request merged**